patches/minecraft/net/minecraft/pathfinding/FlyingNodeProcessor.java.patch

```diff
@@ -0,0 +1,20 @@
+--- ../src-base/minecraft/net/minecraft/pathfinding/FlyingNodeProcessor.java
++++ ../src-work/minecraft/net/minecraft/pathfinding/FlyingNodeProcessor.java
+@@ -272,7 +272,9 @@
+         EnumSet<PathNodeType> enumset = EnumSet.<PathNodeType>noneOf(PathNodeType.class);
+         PathNodeType pathnodetype = PathNodeType.BLOCKED;
+         BlockPos blockpos = new BlockPos(p_186319_5_);
++        this.currentEntity = p_186319_5_;
+         pathnodetype = this.func_193577_a(p_186319_1_, p_186319_2_, p_186319_3_, p_186319_4_, p_186319_6_, p_186319_7_, p_186319_8_, p_186319_9_, p_186319_10_, enumset, pathnodetype, blockpos);
++        this.currentEntity = null;
+ 
+         if (enumset.contains(PathNodeType.FENCE))
+         {
+@@ -321,6 +323,7 @@
+                 {
+                     pathnodetype = PathNodeType.DAMAGE_CACTUS;
+                 }
++                else if (pathnodetype1 == PathNodeType.DAMAGE_OTHER) pathnodetype = PathNodeType.DAMAGE_OTHER;
```

Comment:
What's the purpose of this change?

Reply:
This is the same as the change in `WalkNodeProcessor`, although the decompiling here arranges the logic slightly differently (and a little strangely).

The logic here considers positions to be "damage" if the position directly under them is "damage". The vanilla logic only applies this to "catus" and "fire" damage types, probably because the hardcoded logic in `getPathNodeTypeRaw` doesn't return a type of `DAMAGE_OTHER`. As this type can now be returned by blocks (since #3546), it should be handled correctly as well.